consensus.go

```diff
@@ -674,15 +702,32 @@ func (p *Pbft) runRoundChangeState(ctx context.Context) {
 		// we only expect RoundChange messages right now
 		num := p.state.AddRoundMessage(msg)
 
-		if num == p.state.NumValid() {
-			// start a new round inmediatly
-			p.state.SetCurrentRound(msg.View.Round)
-			p.setState(AcceptState)
-		} else if num == p.state.MaxFaultyNodes()+1 {
-			// weak certificate, try to catch up if our round number is smaller
-			if p.state.GetCurrentRound() < msg.View.Round {
-				// update timer
-				sendRoundChange(msg.View.Round)
+		if p.config.VotingPower == nil {
+			if num == p.state.NumValid() {
+				// start a new round inmediatly
+				p.state.SetCurrentRound(msg.View.Round)
+				p.setState(AcceptState)
+			} else if num == p.state.MaxFaultyNodes()+1 {
+				// weak certificate, try to catch up if our round number is smaller
+				if p.state.GetCurrentRound() < msg.View.Round {
+					// update timer
+					sendRoundChange(msg.View.Round)
+				}
+			}
+		} else {
+			totalVotingPower := totalVotingPower(p.config.VotingPower)
+			validVotingPower := QuorumSizeVP(totalVotingPower)
+			roundVotingPower := p.state.calculateMessagesVotingPower(p.state.roundMessages[msg.View.Round], p.config.VotingPower)
+			if roundVotingPower >= validVotingPower {
+				// start a new round inmediatly
+				p.state.SetCurrentRound(msg.View.Round)
+				p.setState(AcceptState)
+			} else if roundVotingPower > MaxFaultyVP(totalVotingPower)+1 {
```

Comment:
This condition should probably be 
`else if roundVotingPower > MaxFaultyVP(totalVotingPower)`
or 
`else if roundVotingPower >= MaxFaultyVP(totalVotingPower)+1`
?

Reply:
Right. Thank you!